### PR TITLE
feat: enhance 404 pages by showing relevant links

### PR DESCRIPTION
**Pull request by marc2332**
# Description of change

Depends on https://github.com/iota-wiki/iota-wiki/pull/802.

Show relevant docs in 404 considering the URL the user was trying to go to but couldn't be found:

image

## Links to any relevant issues

closes https://github.com/boxfish-studio/iota-wiki/issues/1

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement 

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/theme/NotFound.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                <Translate
+                  id='theme.NotFound.title'
+                  description='The title of the 404 page'
+                >
+                  Page Not Found
+                </Translate>
+              </h1>
+              {possibleMatchingEntries.length > 0 ? (
+                <>
+                  <p>Maybe you were looking for these:</p>
```
Comment (begonaalvarezd) [inline src/theme/NotFound.tsx:66]: I would improve this message, thoughts @Dr-Electron ?

**Review by begonaalvarezd — APPROVED** [review]
🔥 nice

File: src/theme/NotFound.tsx (hunk 1 above)
Comment (jlvandenhout) [reply src/theme/NotFound.tsx:66]: Maybe more like:

### This article does not exist in Shimmer

Articles related to the topic you might be interested in:

- ...
- ...

@lucas-tortora @aleksei-korolev Maybe you have thoughts about this?

Comment (lucas-tortora) [reply src/theme/NotFound.tsx:66]: @jlvandenhout I think we could try something a bit more informal, for example:

Comment (Dr-Electron) [reply src/theme/NotFound.tsx:66]: I like the first version of @lucas-tortora. Do we want to try to go with that? 🤔 

```suggestion
                  <p>It seems we could not find what you're looking for, but you may be interested in:</p>
```

**Review by jlvandenhout — APPROVED** [review]

**Pull request merged**